scipy/sparse/csgraph/_flow.pyx

```diff
@@ -75,9 +87,12 @@ def maximum_flow(csgraph, source, sink):
     By the max-flow min-cut theorem, the maximal value of the flow is also the
     total weight of the edges in a minimum cut.
 
-    To solve the problem, we use the Edmonds--Karp algorithm. [1]_ This
-    particular implementation strives to exploit sparsity. Its time complexity
-    is :math:`O(VE^2)` and its space complexity is :math:`O(E)`.
+    To solve the problem, we provide Edmonds--Karp [1]_ and Dinic's algorithm [4]_. 
+    The implementation of former strives to exploit sparsity. Its time complexity
+    is :math:`O(VE^2)` and its space complexity is :math:`O(E)`. The latter 
+    achieves its performance by building level graphs and finding blocking flows
+    in them.  Its time complexity is :math:`O(V^2E)` and its space complexity is 
+    :math:`O(E)`.
```

Comment:
A small suggestion for consistency regarding the subsequent notation used then in docstrings.
```suggestion
    To solve the problem, we provide Edmonds--Karp [1]_ and Dinic's algorithm [4]_. 
    The implementation of former strives to exploit sparsity. Its time complexity
    is :math:`O(|V|\,|E|^2)` and its space complexity is :math:`O(|E|)`. The latter 
    achieves its performance by building level graphs and finding blocking flows
    in them.  Its time complexity is :math:`O(|V|^2\,|E|)` and its space complexity is 
    :math:`O(|E|)`.
```